src/Compilers/Core/Portable/Collections/ArrayBuilderExtensions.cs

```diff
@@ -249,5 +250,26 @@ public static void FreeAll<T>(this ArrayBuilder<T> builder, Func<T, ArrayBuilder
             }
             builder.Free();
         }
+
+#if COMPILERCORE
+
+        /// <summary>
+        /// Realizes the OneOrMany and disposes the builder in one operation.
+        /// </summary>
+        public static OneOrMany<T> ToOneOrManyAndFree<T>(this ArrayBuilder<T> builder)
+        {
+            if (builder.Count == 1)
+            {
+                var result = OneOrMany.Create(builder[0]);
+                builder.Free();
+                return result;
+            }
+            else
+            {
+                return OneOrMany.Create(builder.ToImmutableAndFree());
```

Comment:
I wonder if `OneOrMany` could be implemented similar to [`StringValues`](https://source.dot.net/#Microsoft.Extensions.Primitives/StringValues.cs,17)  with just one field holding either a `T` or `T[]` value?

Reply:
I considered taht, but i didn't want to potentially regress things by adding all the type-checks.  I leave that as something the compiler team can look into if they want.